src/main/java/bowling/domain/ShotScores.java

```diff
@@ -1,46 +1,70 @@
 package bowling.domain;
 
-import bowling.dto.ShotScoreDto;
-
+import java.util.ArrayList;
 import java.util.List;
-import java.util.stream.Collectors;
+import java.util.Objects;
+import java.util.stream.Stream;
 
 class ShotScores {
     private final List<ShotScore> shotScores;
 
-    ShotScores(List<ShotScore> shotScores) {
-        this.shotScores = shotScores;
+    private ShotScores(List<ShotScore> shotScores) {
+        this.shotScores = new ArrayList<>(shotScores);
     }
 
-    private ShotScore getLast() {
-        return shotScores.get(shotScores.size() - 1);
+    static ShotScores of(List<ShotScore> shotScores) {
+        return new ShotScores(shotScores);
     }
 
-    void add(int shotScore, boolean canShootingBonus) {
-        if (shotScores.isEmpty() || checkCanShootingBonus(canShootingBonus)) {
-            shotScores.add(ShotScore.of(shotScore));
-            return;
-        }
-
-        shotScores.add(this.getLast().next(shotScore));
+    void add(ShotScore shotScore) {
+        shotScores.add(shotScore);
     }
 
-    private boolean checkCanShootingBonus(boolean hasBonusShot) {
-        return this.getLast().isClear() && hasBonusShot;
+    ShotScore getNext(int shot) {
+        if (shotScores.isEmpty()) {
+            return ShotScore.init(shot);
+        }
+        return shotScores.get(shotScores.size() - 1).next(shot);
     }
 
     boolean isSize(int size) {
```

Comment:
메서드 이름은 `hasSize`가 적절해 보여요.

```suggestion
    boolean hasSize(int size) {
```

Reply:
👌